std/machines/small_field/pointer_arith.asm

```diff
@@ -0,0 +1,38 @@
+/// Creates constraints that increments a pointer by amount.
+///
+/// Preconditions:
+/// - amount must be a power of two,
+/// - amount must be in range [1, 0x10000],
+/// - pre_low must be a multiple of amount,
+/// - pre_low must be in range [0, 0xffff].
+///
+/// The pointers are given in 2 16-bit limbs. This function returns a set of constraints
+/// ensuring that the following are equal:
```

Comment:
Should we also say something about the range of the post limbs?

Reply:
Should we? I don't think it is needed.

I think the description as 2 16-bit limbs makes it clear.

I'll add a precondition that the arithmetic should not overflow u32::MAX.